src/api/base.ts

```diff
@@ -68,58 +32,127 @@ const apiAxiosInstance = axios.create({
 // apply request intercepts
 apiAxiosInstance.interceptors.request.use(apiDelayReqIntercept)
 
-// apply response intercepts
-apiAxiosInstance.interceptors.response.use(apiDataRespIntercept)
-
-const BaseAPI = {
-  makeRequest<T = any>(
-    config: AxiosRequestConfig,
-    opts: RequestOptions
-  ): Promise<T> {
-    const wait = window.setTimeout(() => {
-      if (!opts.skipLoader) {
-        useAppSpinner.show()
+export const httpRequest = <T = any>(
+  config: AxiosRequestConfig,
+  opts: RequestOptions
+): HttpPromise<T> => {
+  const options: RequestOptions = {
+    skipLoader: false,
+    withDelay: 0,
+    ...opts,
+  }
+
+  const wait = window.setTimeout(() => {
+    if (options.skipLoader !== true) {
+      useAppSpinner.show()
+    }
+  }, 200)
+
+  return apiAxiosInstance({ ...config, ...opts })
+    .then((success: AxiosResponse<T>) => {
+      return success.data
+    })
+    .catch((err: unknown) => {
+      if (axios.isAxiosError(err)) {
+        throw new HttpError(
+          err.message,
+          err.response?.status,
+          err.response?.data,
+          axios.isCancel(err) ? HTTP_CANCELLED_ERROR : HTTP_ERROR
+        )
       }
-    }, 200)
-
-    return apiAxiosInstance({ ...config, ...opts })
-      .then((success) => success.data)
-      .finally(() => {
-        if (!opts.skipLoader) useAppSpinner.hide()
-        window.clearTimeout(wait)
-      })
-  },
+
+      if (err instanceof Error) {
+        throw new HttpError(err.message, 0)
+      }
+
+      throw new HttpError("An unknown error has occurred.", 0)
+    })
+    .finally(() => {
+      if (options.skipLoader !== true) {
+        useAppSpinner.hide()
+      }
+
+      window.clearTimeout(wait)
+    })
+}
+
+const BaseAPI: HttpClient = {
   get<T = any>(
     path: string,
-    opts: RequestOptions,
+    opts?: RequestOptions,
```

Comment:
I felt like this should be optional or have a default `{}`.  Consumers shouldn't need to provide an empty object as a default.